diff --git a/BDD/bdd_external.h b/BDD/bdd_external.h
index 8cb60a5..b822d0d 100644
--- a/BDD/bdd_external.h
+++ b/BDD/bdd_external.h
@@ -21,6 +21,8 @@
 #ifndef __BDD_EXTERNAL_H
 #define __BDD_EXTERNAL_H
 
+#define export _export
+
 #include "bdd.h"
 
 typedef struct _BddNode {
diff --git a/BDD/makefile.am b/BDD/makefile.am
index 3acf219..f73a77d 100644
--- a/BDD/makefile.am
+++ b/BDD/makefile.am
@@ -6,4 +6,4 @@ libmonabdd_la_SOURCES = bdd.c bdd_cache.c bdd_double.c bdd_dump.c bdd_dump.h \
 libmonabdd_la_LDFLAGS = -version-info 1:4
 libmonabdd_la_LIBADD = ../Mem/libmonamem.la
 monadir = $(includedir)/mona
-mona_HEADERS = bdd.h
+mona_HEADERS = bdd.h bdd_external.h bdd_dump.h
diff --git a/DFA/makebasic.c b/DFA/makebasic.c
index ec2f578..bbc40ce 100644
--- a/DFA/makebasic.c
+++ b/DFA/makebasic.c
@@ -24,8 +24,8 @@
 #include "dfa.h"
 #include "../BDD/bdd_internal.h"
 
-#define MAX_EXCEPTION 50
-#define MAX_VARIABLES 10
+#define MAX_EXCEPTION 1048576
+#define MAX_VARIABLES 2000
 /* #warning INTERNAL LIMITS */
 
 struct path_descr {
