diff --git a/BDD/bdd.h b/BDD/bdd.h
index fc80c47..71f4799 100644
--- a/BDD/bdd.h
+++ b/BDD/bdd.h
@@ -18,6 +18,10 @@
  * USA.
  */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #ifndef _BDD_PACKAGE
 #define _BDD_PACKAGE
 
@@ -414,3 +418,7 @@ extern unsigned fn_identity(unsigned p);
   abort(), 0)))
 
 #endif
+
+#ifdef __cplusplus
+}
+#endif
diff --git a/BDD/bdd_dump.h b/BDD/bdd_dump.h
index 47fb409..941e262 100644
--- a/BDD/bdd_dump.h
+++ b/BDD/bdd_dump.h
@@ -17,6 +17,9 @@
  * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335,
  * USA.
  */
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 #ifndef __BDD_DUMP_H
 #define __BDD_DUMP_H
@@ -28,3 +31,8 @@ void bddDumpNode(bdd_manager *bddm, bdd_ptr p);
 void bddDump(bdd_manager *bddm);
 
 #endif
+
+#ifdef __cplusplus
+}
+#endif
+
diff --git a/BDD/bdd_external.h b/BDD/bdd_external.h
index 8cb60a5..5da6396 100644
--- a/BDD/bdd_external.h
+++ b/BDD/bdd_external.h
@@ -17,10 +17,15 @@
  * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335,
  * USA.
  */
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 #ifndef __BDD_EXTERNAL_H
 #define __BDD_EXTERNAL_H
 
+#define export _export
+
 #include "bdd.h"
 
 typedef struct _BddNode {
@@ -43,3 +48,8 @@ void export(bdd_manager *bddm, unsigned p, Table *table);
 unsigned make_node(int i);
 
 #endif
+
+#ifdef __cplusplus
+}
+#endif
+
diff --git a/BDD/makefile.am b/BDD/makefile.am
index 3acf219..f73a77d 100644
--- a/BDD/makefile.am
+++ b/BDD/makefile.am
@@ -6,4 +6,4 @@ libmonabdd_la_SOURCES = bdd.c bdd_cache.c bdd_double.c bdd_dump.c bdd_dump.h \
 libmonabdd_la_LDFLAGS = -version-info 1:4
 libmonabdd_la_LIBADD = ../Mem/libmonamem.la
 monadir = $(includedir)/mona
-mona_HEADERS = bdd.h
+mona_HEADERS = bdd.h bdd_external.h bdd_dump.h
diff --git a/DFA/dfa.h b/DFA/dfa.h
index 4ca7a86..47d3047 100644
--- a/DFA/dfa.h
+++ b/DFA/dfa.h
@@ -20,6 +20,10 @@
 
 /* See the 'MONA User Manual' for documentation */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #ifndef __DFA_H
 #define __DFA_H
 
@@ -123,3 +127,7 @@ DFA *dfaMinusModulo1(int i, int j, int k);
 DFA *dfaPresbConst(int i, int n);
 
 #endif
+
+#ifdef __cplusplus
+}
+#endif
diff --git a/DFA/makebasic.c b/DFA/makebasic.c
index ec2f578..bbc40ce 100644
--- a/DFA/makebasic.c
+++ b/DFA/makebasic.c
@@ -24,8 +24,8 @@
 #include "dfa.h"
 #include "../BDD/bdd_internal.h"
 
-#define MAX_EXCEPTION 50
-#define MAX_VARIABLES 10
+#define MAX_EXCEPTION 1048576
+#define MAX_VARIABLES 2000
 /* #warning INTERNAL LIMITS */
 
 struct path_descr {
diff --git a/Mem/mem.h b/Mem/mem.h
index 37bdbdf..fcdd35f 100644
--- a/Mem/mem.h
+++ b/Mem/mem.h
@@ -17,6 +17,9 @@
  * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335,
  * USA.
  */
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 #ifndef __MEM_H
 #define __MEM_H
@@ -39,3 +42,7 @@ void mem_zero(void *, size_t);
 unsigned int mem_allocated();
 
 #endif
+
+#ifdef __cplusplus
+}
+#endif
